opm/simulators/wells/WellGroupHelpers.cpp

```diff
@@ -897,7 +900,12 @@ namespace WellGroupHelpers
 #endif
                 } else {
                     // Table number specified as 9999 in the deck, no pressure loss.
-                    node_pressures[node] = up_press;
+                    if (network.node(node).as_choke()){
+                        // PJPE: Node pressure is set to the common THP of the wells
+                        node_pressures[node] = group_state.well_group_thp(node);
```

Comment:
as mentioned in the private communication, the nodal pressure might not be the common THP of the wells. The nodal pressure is still calculated from the branch VFP calculation. We **might** need a variable to store the choke pressure.  Or the difference between the nodal pressure and the common THP constraints will be the choke pressure, then we do not need to store it explicitly.